src/backend/distributed/commands/utility_hook.c

```diff
@@ -224,10 +224,20 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 										   params, queryEnv, dest, completionTag);
 
 			StoredProcedureLevel -= 1;
+
+			if (InDelegatedProcedureCall && StoredProcedureLevel == 0)
+			{
+				InDelegatedProcedureCall = false;
+			}
 		}
 		PG_CATCH();
 		{
 			StoredProcedureLevel -= 1;
+
+			if (InDelegatedProcedureCall && StoredProcedureLevel == 0)
```

Comment:
Thinking out loud, what will happen in case of nested stored procedure level? Since we rethrow, at some point we have to have StoredProcedureLevel  to zero?

Reply:
right, we'll go through all the PG_CATCH sections